### PR TITLE
Add initial support for janet-lang

### PR DESCRIPTION
**Pull request by pnelson**
Resolves https://github.com/helix-editor/helix/issues/4617 https://github.com/helix-editor/helix/issues/4574

### CODE REVIEW

File: languages.toml [hunk 1]
```diff
@@ -2952,3 +2952,13 @@ file-types = ["log"]
...
 source = { git = "https://github.com/Tudyx/tree-sitter-log", rev = "62cfe307e942af3417171243b599cc7deac5eab9" }
+
+[[language]]
+name = "janet"
+scope = "source.janet"
+injection-regex = "janet"
+file-types = ["janet"]
+comment-token = "#"
+indent = { tab-width = 2, unit = "  " }
+roots = []
```
Comment (the-mikedavis) [inline languages.toml:2963]: This key can be omitted since https://github.com/helix-editor/helix/pull/8803

**Review by the-mikedavis — APPROVED** [review]

**Comment by mikkelfj** [comment]
Great to see this, but as far as I can tell, clojure-lsp listed in the PR requires openjdk installed on macos, which completely defeats the point of using janet in the first place.

**Comment by the-mikedavis** [comment]
This PR doesn't configure a language server, it's just re-using the highlighting from clojure. If there's a language server for janet, that could be added in a follow-up PR

**Comment by mikkelfj** [comment]
Yes, I just realized. Sorry for the noise.

**Comment by sogaiu** [comment]
FWIW, I have a [branch](https://github.com/sogaiu/helix/tree/janet) based on a helix from 2023-10 or so that uses [tree-sitter-janet-simple](https://github.com/sogaiu/tree-sitter-janet-simple).  The branch also includes a highlighting query file that works with the grammar.

I was waiting for lsp support to improve, but perhaps that can be added later?

Please let me know if there is any interest.

**Comment by pnelson** [comment]
> Please let me know if there is any interest.

Absolutely! :rocket:

**Comment by sogaiu** [comment]
Ok, I've created a [new branch](https://github.com/sogaiu/helix/tree/janet-simple) that's based on 7e389b67c24dfe4466112c988b240c807e7e2414.

Before I make a PR, I have a few questions:

* I'd chosen [`janet-simple` for `name`](https://github.com/sogaiu/helix/commit/5e795df06675840c3e791e657ea66592a535cea8#diff-f3ea26d68561619ec2cb9ee551ef03916b667bd5f55b2ae2c54cae96e8821b2eR328) before [1] and I see that this PR used `janet`.  My choice was influenced by the grammar name.  Is this ok or should it be changed to use `janet`?
* I see #9081 has something for `injection-regex` -- is this seen as important?

Hmm...perhaps better to make a PR and ask there?

---

[1] There is more than one tree-sitter grammar for Janet -- though the other one I know of is not in use in Emacs or Neovim or elsewhere -- it took a different approach and possibly some folks might want to use it for certain situations.  They were started around the same time but I changed the name of mine to avoid name collisions.

**Comment by rokf** [comment]
I've tried https://github.com/CFiggers/janet-lsp today and it's working. I've tested this on macOS with Janet installed through `brew`. It doesn't have many features yet though, but it could be a start 🙂

First I had to make sure that the location where binaries that are installed through `jpm` are placed was in my path, i.e.:

```sh
export PATH=/opt/homebrew/Cellar/janet/HEAD-2f0c789/bin:$PATH
```

I've (somewhat) followed the installation instructions from the README:

```sh
git clone https://github.com/CFiggers/janet-lsp
cd janet-lsp
jpm -l deps
jpm install
```

This is what I've put into `languages.toml` in my `~/.config/helix` folder:


```toml
[language-server.janet-lsp]
command = "janet-lsp"

[[language]]
name= "janet"
language-servers = [ { name = "janet-lsp" } ]
```

**Pull request merged**